docker/docker.go

```diff
@@ -101,6 +105,17 @@ func New() (*docker, error) {
 func (d *docker) CollectMetrics(mts []plugin.Metric) ([]plugin.Metric, error) {
 	var err error
 	metrics := []plugin.Metric{}
+
+	conf, err := getDockerConfig(mts[0])
+	if err != nil {
+		return nil, err
+	}
+
+	err = d.init_client(conf["endpoint"])
```

Comment:
This way you are going to initialize new client for each call to `CollectMetrics`. I would suggest to do it only once. (check for `nil`?)